src/views/ProtestPage.js

```diff
@@ -68,7 +94,25 @@ function useFetchProtest() {
   };
 }
 
-function ProtestPageContent({ protest, canEdit }) {
+function shouldShowEditButton({ protestData, user }) {  
+  if (!user) {
+    return false;
+  }
+
+  if (isAdmin(user)) {
+    return true;
+  }
+
+  // This will only show the button in case there is no leader at the moment
+  // or in case you are the leader
+  if (!protestData?.roles?.leader || protestData?.roles?.leader?.includes(user.uid)) {
+    return true;
+  } 
```

Comment:
We should consider removing this, we can have multiple leaders.

Reply:
I think we discussed with @guytepper and for now will leave this. could be changed later on